x-pack/plugins/watcher/__jest__/client_integration/watch_list.test.ts

```diff
@@ -76,12 +78,68 @@ describe('<WatchList />', () => {
         beforeEach(async () => {
           httpRequestsMockHelpers.setLoadWatchesResponse({ watches });
 
-          testBed = await setup();
+          await act(async () => {
+            testBed = await setup();
+          });
+
+          testBed.component.update();
+        });
+
+        test('should retain the search query', async () => {
+          const { find, component, table } = testBed;
+
+          const searchInput = find('watchesTableContainer').find('.euiFieldSearch');
+
+          // Enter the name of "watch1" in the search box
+          // @ts-ignore
+          searchInput.instance().value = watch1.name;
+          searchInput.simulate('keyup', { key: 'Enter', keyCode: 13, which: 13 });
+
+          component.update();
+
+          const { tableCellsValues } = table.getMetaData('watchesTable');
+
+          // Expect "watch1" is only visible in the table
+          expect(tableCellsValues.length).toEqual(1);
+          const row = tableCellsValues[0];
+          const { name, id, watchStatus } = watch1;
+
+          const getExpectedValue = (value: any) => (typeof value === 'undefined' ? '' : value);
+
+          expect(row).toEqual([
+            '',
+            id,
+            getExpectedValue(name),
+            watchStatus.state,
+            getExpectedValue(watchStatus.comment),
+            getExpectedValue(watchStatus.lastMetCondition),
+            getExpectedValue(watchStatus.lastChecked),
+            '',
+          ]);
 
           await act(async () => {
-            await nextTick();
-            testBed.component.update();
+            // Advance timers to simulate another request
```

Comment:
Similar DX suggestion here about a helper function. We can add this to `watch_list.helpers.ts`:

```ts
  const advanceTimeToTableRefresh = async () => {
    const { component } = testBed;
    await act(async () => {
      // Advance timers to simulate another request
      jest.advanceTimersByTime(REFRESH_INTERVALS.WATCH_LIST);
    });
    component.update();
  };

  return {
    ...testBed,
    actions: {
      selectWatchAt,
      clickWatchAt,
      clickWatchActionAt,
      advanceTimeToTableRefresh,
    },
  };
```

And use it here:

```ts
await advanceTimeToTableRefresh();
```